packages/api-client/src/api/getProduct/index.ts

```diff
@@ -2,14 +2,18 @@ import { ApiContext } from '../../types';
 import { addHostToProductImages, deserializeSingleProductVariants } from '../serializers/product';
 
 export default async function getProduct({ client, config }: ApiContext, { slug }) {
+  const variantFields = 'name,description,slug,sku,price,display_price,in_stock,product,images,option_values,is_master';
+
   const result = await client.products.show(
     slug,
     {
-      include: 'variants.option_values,option_types,product_properties,taxons,taxons.parent,images',
       fields: {
-        product: 'name,variants,option_types,product_properties,taxons',
-        variant: 'name,description,slug,sku,price,display_price,in_stock,product,images,option_values'
-      }
+        product: 'name,primary_variant,default_variant,variants,option_types,product_properties,taxons',
+        variant: variantFields,
+        primary_variant: variantFields,
```

Comment:
@tniezg @damianlegawiec what's the difference between primary_variant and default_variant?

Reply:
@rafalcymerys primary is the old master variant (we're changing master to primary naming in the future)
default variant is the first non-primary variant in stock, if there's only primary (master) present, the primary == default

Reply:
Gotcha, I wasn't sure about the primary_variant.
Will this query work with older Spree versions e.g. Spree 4.2, if we add includes that weren't there?

I'm aiming at having compatibility with at least 4.2, since it has almost all the features required by VSF, except some minor things. This could help us in speeding-up the adoption of VSF & Spree combo without adding another obstacle of needing to migrate to 4.3 (which we would of course aim for, but just keeping the barier to entry low).

I created a definition for `spreeFeatures` which you can see in use in `packages/api-client/src/api/logIn/index.ts:26`. This is pretty much a feature flag that lets us enable some newer features on demand. Maybe primary_variant would also be a good candidate for it?

Reply:
@rafalcymerys we can backport the primary variant addition to 4.2 branch

Reply:
Right, but then users on 4.2 would still need to upgrade to 4.2.6, no plug-and-play magic ;)
I can add this feature flag later though, this should be a simple change.